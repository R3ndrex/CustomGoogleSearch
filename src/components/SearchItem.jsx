export default function SearchItem({ item }) {
    return (
        <div className="search-item">
            <h2>
                <a href={item.link} target="_blank">
                    {item.title}
                </a>
            </h2>
            <p className="display-link">{item.displayLink}</p>
            <p>{item.snippet}</p>
        </div>
    );
}
