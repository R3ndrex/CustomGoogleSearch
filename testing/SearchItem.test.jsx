import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import SearchItem from "../src/components/SearchItem";

const item = {
    title: "Title",
    link: "test-link/test",
    displayLink: "test-link",
    snippet: "bla bla bla",
};

describe("Search Item", () => {
    it("renders correctly", () => {
        const { container } = render(<SearchItem item={item} />);
        expect(container).toMatchSnapshot();
    });
});
