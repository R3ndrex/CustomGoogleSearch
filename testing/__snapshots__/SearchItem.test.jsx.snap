// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search Item > renders correctly 1`] = `
<div>
  <div
    class="search-item"
  >
    <h2>
      <a
        href="test-link/test"
        target="_blank"
      >
        Title
      </a>
    </h2>
    <p
      class="display-link"
    >
      test-link
    </p>
    <p>
      bla bla bla
    </p>
  </div>
</div>
`;
